src/openrct2/config/Config.cpp

```diff
@@ -798,13 +811,61 @@ bool config_find_or_browse_install_directory()
             {
                 auto uiContext = GetContext()->GetUiContext();
                 uiContext->ShowMessageBox(format_string(STR_NEEDS_RCT2_FILES, nullptr));
-
-                std::string installPath = uiContext->ShowDirectoryDialog(format_string(STR_PICK_RCT2_DIR, nullptr));
+                std::string gog = language_get_string(STR_OWN_ON_GOG);
+                std::string hdd = language_get_string(STR_INSTALLED_ON_HDD);
+
+                std::vector<std::string> options;
+                std::string chosenOption;
+                // so far, only Linux has ShowMenuDialog implemented
+#if (defined(__unix__) || defined(__EMSCRIPTEN__)) && !defined(__ANDROID__) && !defined(__APPLE__)
```

Comment:
This should be a platform function. It could be named `HasMenuSupport` and return a bool.